### PR TITLE
Don't require .packages file for `flutter packages` command

### PR DESCRIPTION
**Pull request by tvolkert**
(it's the command that populates the .packages file)

### CODE REVIEW

**Comment by tvolkert** [comment]
Tests to follow, but sending this out to unblock broken builds on other repos.

**Review by yjbanov — APPROVED** [review]
LGTM

**Comment by mit-mit** [comment]
Thanks!

File: packages/flutter_tools/lib/src/runner/flutter_command.dart [hunk 1]
```diff
@@ -347,7 +347,7 @@ abstract class FlutterCommand extends Command<Null> {
       }
 
       // Validate the current package map only if we will not be running "pub get" later.
-      if (!(_usesPubOption && argResults['pub'])) {
+      if (parent?.name != 'packages' && !(_usesPubOption && argResults['pub'])) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/runner/flutter_command.dart]: test?

**Comment by tvolkert** [comment]
> test?

See comment above.  Writing tests as we speak, but I wanted to unbreak the build first.

**Pull request merged**